### PR TITLE
feat(specification): calculate features application logic

### PR DESCRIPTION
**Pull request by kgajowy**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/apps/api/src/modules/specification/calculate-features/application/calculate-features-for-specification.handler.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    featureIds,
+    specificationId,
+  }: CalculateFeaturesForSpecification): Promise<void> {
+    let specification = await this.specificationRepository.getById(
+      specificationId,
+    );
+    if (!specification) {
+      this.logger.warn(`Couldn't find specification: ${specificationId}`);
+      return;
+    }
```
Comment (Dyostiq) [inline api/apps/api/src/modules/specification/calculate-features/application/calculate-features-for-specification.handler.ts:34]: I'd say it's an error

File: api/apps/api/src/modules/specification/calculate-features/application/specification.repository.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Specification } from '../../specification';
+
+export abstract class SpecificationRepository {
+  abstract save(specification: Specification): Promise<void>;
+
+  abstract getById(id: string): Promise<Specification | undefined>;
```
Comment (Dyostiq) [inline api/apps/api/src/modules/specification/calculate-features/application/specification.repository.ts:6]: it's a lot of duplication having a multiple specification repositories...

no extremes are healthy :D

Comment (kgajowy) [reply api/apps/api/src/modules/specification/calculate-features/application/specification.repository.ts:6]: If it happens (seems like it does) they will be 100% same, will extract them once all application pieces are in place.

Comment (kgajowy) [reply api/apps/api/src/modules/specification/calculate-features/application/specification.repository.ts:6]: fixed in https://github.com/Vizzuality/marxan-cloud/pull/417/files via https://github.com/Vizzuality/marxan-cloud/pull/417/commits/94946b083a4e324560dcb1e72bdd94f5f4b46799

File: api/apps/api/src/modules/specification/calculate-features/application/calculate-features-for-specification.handler.ts (hunk 1 above)
Comment (kgajowy) [reply api/apps/api/src/modules/specification/calculate-features/application/calculate-features-for-specification.handler.ts:34]: fixed in https://github.com/Vizzuality/marxan-cloud/pull/417 via 
* https://github.com/Vizzuality/marxan-cloud/pull/417/commits/b0a568064ca61b4f26ea5aa3d4e3fe0cf6e39563
* https://github.com/Vizzuality/marxan-cloud/pull/417/commits/9b8b6dd9e1d20e8c91c1f5ba739f215820a4678d

**Pull request merged**